### PR TITLE
A modified Kim CNN that has embedding moved out

### PR DESCRIPTION
**Pull request by l1m2p3**
1. move embedding outside the model (see `model.py` and `embedding.py`)
2. do embedding before passing a batch into the model (see 'train.py')

The rest of the file should be the same as Peng's implementation, including how to run/train

### CODE REVIEW

File: kim_cnn/test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import torch
```
Comment (tuzhucheng) [inline kim_cnn/test.py]: Please replace all tabs with 4 spaces in this file.

File: kim_cnn/train.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                    break
+
+        if iterations % args.log_every == 1:
+            # print progress message
+            print(log_template.format(time.time() - start,
+                                      epoch, iterations, 1 + batch_idx, len(train_iter),
+                                      100. * (1 + batch_idx) / len(train_iter), loss.data[0], ' ' * 8,
+                                      n_correct / n_total * 100, ' ' * 12))
+
+
```
Comment (tuzhucheng) [inline kim_cnn/train.py]: Please delete these extra lines.

File: kim_cnn/get_data.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+mkdir data
```
Comment (tuzhucheng) [inline kim_cnn/get_data.sh:2]: Please make sure this file is executable (use chmod to add it if it is not). Also add shebang: `#!/usr/bin/env bash`.

File: kim_cnn/test.py (hunk 1 above)
Comment (l1m2p3) [reply kim_cnn/test.py]: Ah sorry I missed this file when I was cleaning up. This was a file that I put all the test code in and shouldn't be uploaded. Thanks for spotting it out.

File: kim_cnn/train.py (hunk 2 above)
Comment (l1m2p3) [reply kim_cnn/train.py]: removed

File: kim_cnn/get_data.sh (hunk 3 above)
Comment (l1m2p3) [reply kim_cnn/get_data.sh:2]: added

**Review by tuzhucheng — APPROVED** [review]

**Pull request merged**